narwhals/_duckdb/expr.py

```diff
@@ -728,30 +764,58 @@ def rank(self, method: RankMethod, *, descending: bool) -> Self:
         else:  # method == "ordinal"
             func = FunctionExpression("row_number")
 
-        def _rank(_input: duckdb.Expression) -> duckdb.Expression:
+        def _rank(
+            _input: duckdb.Expression,
+            *,
+            descending: bool,
+            partition_by_sql: str | None = None,
+        ) -> duckdb.Expression:
             if descending:
                 by_sql = f"{_input} desc nulls last"
             else:
                 by_sql = f"{_input} asc nulls last"
             order_by_sql = f"order by {by_sql}"
```

Comment:
it might be clearer for this `order_by_sql` to go in the `else` block (line 784)?